### ISSUE TITLE
[Bug] [Connector-V2][Postgres-CDC] Scanning unwanted database and tables which are not mentioned in Source 

### ISSUE THREAD

**Issue by ratadepally** [apache/seatunnel#8184]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

When we provide a hostname for Postgres-CDC connection and try to run the job we are seeing unwanted WARN logs which says TableDiscoveryUtils class is trying to scan for all the databases under the specified host. Even though we specify the database name in the source config we are still getting these unwanted warnings. Is there any way to avoid other databases and tables which are not part of the connection ?

### SeaTunnel Version

seatunnel-2.3.8

### SeaTunnel Config

```conf
env {
  execution.parallelism = 1
  job.mode = "STREAMING"
  checkpoint.interval = 5000
  read_limit.bytes_per_second=7000000
  read_limit.rows_per_second=400
}

source {
  
  Postgres-CDC {
    result_table_name = "employees"
    username = "postgres"
    password = "password"
    database-names = ["postgres"]
    schema-names = ["public"]
	slot.name = "employeeslot"
    table-names = ["postgres.public.employees"]
    base-url = "jdbc:postgresql://localhost:5432/postgres"
  }
  
}

transform {
}

sink {
	
	Console {
      source_table_name = "employees"
  }
 
}
```

### Running Command

```shell
./bin/seatunnel.sh --config config/postgres-cdc.template --async -n postgrescdcsync
```

### Error Exception

```log
[915470173375299585] 2024-11-30 23:11:41,544 INFO  [.c.s.c.p.u.TableDiscoveryUtils] [hz.main.seaTunnel.task.thread-128] -                                                                              list of available databases is: [postgres, database2, database3]
[915470173375299585] 2024-11-30 23:11:41,544 INFO  [.c.s.c.p.u.TableDiscoveryUtils] [hz.main.seaTunnel.task.thread-128] - Read list of available tables in each database
[915470173375299585] 2024-11-30 23:11:41,606 WARN  [.c.s.c.p.u.TableDiscoveryUtils] [hz.main.seaTunnel.task.thread-128] -                                                                              skipping database 'database2' due to error reading tables: ERROR: cross-database references are not implemented: "database2.information_schema.tables"
  Position: 15
[915470173375299585] 2024-11-30 23:11:41,657 WARN  [.c.s.c.p.u.TableDiscoveryUtils] [hz.main.seaTunnel.task.thread-128] -                                                                              skipping database 'database3' due to error reading tables: ERROR: cross-database references are not implemented: "database3.information_schema.tables"
  Position: 15
```

### Java or Scala Version

Jdk 1.8

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by naive-zhang** [comment]
@hailin0 I'd like to fix it, please assign it to me, thx.

**Comment by ratadepally** [comment]
Hi Team, any update on this please?

**Comment by davidzollo** [comment]
Reopening this issue because the behavior is still worth fixing, and the previous close came from stale automation rather than a resolved implementation.

If anyone would like to contribute, this is a focused CDC connector bug and a good issue to claim. A strong starting point is the Postgres table discovery path: it should respect configured database/schema/table filters early enough to avoid scanning unrelated databases and emitting noisy WARN logs.

If you'd like to take it, please leave a comment here and we can help clarify the expected behavior and validation scope. Reproduction help, debugging notes, and PRs are all appreciated. Thank you kindly to everyone supporting SeaTunnel.

**Comment by DanielLeens** [comment]
Thanks for reporting this.

I checked the current Postgres CDC discovery path, and this looks like a real connector-side bug rather than just harmless noisy logging.

Right now `TableDiscoveryUtils.listTables()` first reads *all* database names from `pg_database`, then it tries to query `"<db>".INFORMATION_SCHEMA.TABLES` for each one, and only after that applies the configured table filter. So a config like `database-names = [\"postgres\"]` does not actually prevent those cross-database probes from happening.

That matches the warnings you posted: the filter is being applied too late in the flow, after the connector has already tried to inspect the other databases.

I do not see a clean config-only switch in the current implementation that avoids this properly. The correct fix direction is to restrict the database iteration before issuing those per-database queries, instead of treating the cross-database failures as expected warnings.

This issue is worth keeping open as a real bug. We've already marked it as `help wanted`, so a focused PR would be very welcome.
